### PR TITLE
Fix distributed modality batching for small splits

### PR DESCRIPTION
**Pull request by zjn20030811**
## Summary
- make `SplitModalitySampler` safe when a dataset contains only one modality
- cyclically pad short modality partitions so every distributed global batch remains divisible by the replica count
- compute sampler length from independently padded modality batches
- add focused regression coverage for single-modality, mixed-modality, and invalid distributed configurations

## Why
The sampler always unpacked a multimodal partition, then indexed its last batch. A language-only or vision-only dataset therefore failed before the first training step. When both partitions were present but each was smaller than the global batch size, the old padding and `total_size` calculation could also produce undersized batches and inconsistent `__len__` values. This affects VLA fine-tuning runs that use filtered or highly imbalanced modality mixtures.

## Validation
- `python -m pytest -q tests/test_batching_utils.py` (7 passed)
- `python -m py_compile prismatic/util/batching_utils.py tests/test_batching_utils.py`
- `git diff --check`

### CODE REVIEW

**Comment by zjn20030811** [comment]
System.Collections.Hashtable[OpenHelix-Team/VLA-Adapter#65]
